src/libraries/System.Linq/src/System.Linq.csproj

```diff
@@ -1,98 +1,85 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>$(NetCoreAppCurrent);$(NetCoreAppCurrent)-browser;$(NetCoreAppCurrent)-wasi;$(NetCoreAppCurrent)-android;$(NetCoreAppCurrent)-ios;$(NetCoreAppCurrent)-tvos</TargetFrameworks>
+    <TargetFramework>$(NetCoreAppCurrent)</TargetFramework>
     <UseCompilerGeneratedDocXmlFile>false</UseCompilerGeneratedDocXmlFile>
   </PropertyGroup>
 
-  <!-- DesignTimeBuild requires all the TargetFramework Derived Properties to not be present in the first property group. -->
-  <PropertyGroup>
-    <TargetPlatformIdentifier>$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)'))</TargetPlatformIdentifier>
-    <OptimizeForSize Condition="'$(TargetPlatformIdentifier)' == 'browser' or '$(TargetPlatformIdentifier)' == 'android' or '$(TargetPlatformIdentifier)' == 'ios' or '$(TargetPlatformIdentifier)' == 'tvos'">true</OptimizeForSize>
```

Comment:
`OptimizeForSize` was set true for some targets but it's not anymore. 
`WasmFeatures.props` set the default for wasm workload, but default runtime pack also needs to be optimized for size - when people don't use workload.

Reply:
> `OptimizeForSize` was set true for some targets but it's not anymore.

Yes, that is the purpose of this PR. To only have one build of this library (https://github.com/dotnet/runtime/pull/111743#discussion_r1932422189)

> `WasmFeatures.props` set the default for wasm workload, but default runtime pack also needs to be optimized for size - when people don't use workload in Blazor.

Could you please point me to which .targets we need to update? Is it the Microsoft.NET.Sdk.BlazorWebAssembly.6_0.targets linked from https://github.com/dotnet/runtime/pull/111743#issuecomment-2618900157?

Reply:
(To be clear, the OptimizeForSize property controls whether we build System.Linq assembly in a not-fully-compatible manner (see the tests I'm touching where it behaves differently) that has better size characteristics. This PR deletes that build and changes it to always build the compatible Linq implementation. It introduces a feature switch that allows switching to the not-fully-compatible mode. The default for the feature switch is disabled. It can be enabled where the tradeoff is worth it (I have PRs out enabling it in places, linked above). The advantage is that now if a customer runs into the incompatibility, they can just set a property and unblock themselves.)

Reply:
I think maybe we need both 

https://github.com/dotnet/sdk/blob/b89ea044c3ac0e028228e73abcc8310b101b63b8/src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.6_0.targets#L43-L50

 @maraf 
I hope that we run ILLink during publish when wasm-workload is not installed ?
Is there better place ?

and for workload

https://github.com/dotnet/runtime/blob/f1901a00dfcfbcf0a4f8b6aa7459fec30846d1ab/src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.Current.Manifest/WasmFeatures.props

Reply:
Thank you! Folded the BlazorWebAssembly.targets change into https://github.com/dotnet/sdk/pull/46375.

WasmFeatures.props change is already part of the PR here.